src/fhirtypes/common.ts

```diff
@@ -205,6 +213,41 @@ export function replaceField(
   }
 }
 
+/**
+ * Adds Mixin rules onto a Profile, Extension, or Instance
+ * @param {Profile | Extension | Instance} fshDefinition - The definition to apply mixin rules on
+ * @param {FSHTank} tank - The FSHTank containing the fshDefinition
+ */
+export function applyMixinRules(
+  fshDefinition: Profile | Extension | Instance,
+  tank: FSHTank
+): void {
+  // Rules are added to beginning of rules array, so add the last mixin rules first
+  [...fshDefinition.mixins].reverse().forEach(mixinName => {
+    const ruleSet = tank.fish(mixinName, Type.RuleSet) as RuleSet;
+    if (ruleSet) {
+      ruleSet.rules.forEach(r => {
+        // Record source information of Profile/Extension/Instance on which Mixin is applied
+        r.sourceInfo.appliedFile = fshDefinition.sourceInfo.file;
+        r.sourceInfo.appliedLocation = fshDefinition.sourceInfo.location;
+      });
+      const rules = ruleSet.rules.filter(r => {
+        if (fshDefinition instanceof Instance && !(r instanceof FixedValueRule)) {
+          logger.error(
+            'Mixin rules applied to an instance must fix a value. Other rules are ignored.',
```

Comment:
Honest question -- should we refer to these as _Mixin rules_ or _Mixins rules_ (since that's the keyword)?  I'm not sure of the right answer.  Did you think about it?

Reply:
I did think about it, and but nothing really satisfied me. I know `Mixins` is the keyword, but that is the keyword because it is the plural of `Mixin`, and I think `Mixin` is better grammar. I think I will change it to "Rules applied by mixins". Better?

Reply:
Sure!